host-controller/src/main/java/org/jboss/as/host/controller/ServerInventoryImpl.java

```diff
@@ -720,4 +765,36 @@ public void handle(Callback[] callbacks) throws IOException, UnsupportedCallback
             }
         };
     }
-}
+
+    private ModelNode appendServerNameToFailureResponse(String serverName, ModelNode failureResponse) {
+        String currentDescription = failureResponse.get(FAILURE_DESCRIPTION).asString();
+        return getFailureResponse(String.format("%s server: %s", currentDescription, serverName));
+    }
+
+    private ModelNode getTimedOutAwaitingSuspendResponse(int timeout, String serverName) {
+        String msg = HostControllerLogger.ROOT_LOGGER.timedOutAwaitingSuspendResponseMsg(timeout, serverName);
+        return getFailureResponse(msg);
+    }
+
+    private ModelNode getInterruptedAwaitingSuspendResponse(String serverName) {
+        String msg = HostControllerLogger.ROOT_LOGGER.interruptedAwaitingSuspendResponseMsg(serverName);
+        return getFailureResponse(msg);
+    }
+
+    private ModelNode getSuspendExecutionFailedResponse(String serverName) {
+        String msg = HostControllerLogger.ROOT_LOGGER.suspendExecutionFailedMsg(serverName);
+        return getFailureResponse(msg);
+    }
+
+    private ModelNode getSuspendListenerFailedResponse(String serverName) {
+        String msg = HostControllerLogger.ROOT_LOGGER.suspendListenerFailedMsg(serverName);
+        return getFailureResponse(msg);
+    }
+
+    private ModelNode getFailureResponse(String msg){
+        final ModelNode response = new ModelNode();
+        response.get(OUTCOME).set(FAILED);
+        response.get(FAILURE_DESCRIPTION).set(msg);
+        return response;
```

Comment:
The outcome and failure-description bits are not part of what this logic should do; the OperationContext handles that stuff. This could just be 

return new ModelNode(msg);

which can of course be inlined in the calling methods.